uber/site_sections/registration.py

```diff
@@ -142,6 +142,15 @@ def change_badge(self, session, id, message='', **params):
             except ValueError:
                 attendee.badge_num = None
 
+            if attendee.badge_type != c.CHILD_BADGE and old_badge_type == c.CHILD_BADGE:
+                day = c.EPOCH.date() if date.today() <= c.EPOCH.date() else sa.localized_now().date()
+                attendee_age = (day - attendee.birthdate).days // 365.2425
+                reduce_years_by = 18-attendee_age
+                if reduce_years_by > 0.0:
+                    attendee.birthdate = attendee.birthdate\
+                        .replace(year=int(attendee.birthdate.year - reduce_years_by))
+                    attendee.ribbon = c.NO_RIBBON
+
```

Comment:
this is pretty janked up :)    I'm not sure how else to handle this though based on what you and I talked about. @kitsuta  might have input.

2 things: 
1) inline comment here that this is hacking around a specific case and make it clear that the birthday is going to be bogus
2) propagate a message through here to the admin that the birthdate is now wrong and needs to be adjusted